static/panes/tool.ts

```diff
@@ -0,0 +1,570 @@
+        button.addClass('active');
+        this.resize();
+    }
+
+    hidePanel(button, panel) {
+        panel.addClass('d-none');
+        button.removeClass('active');
+        this.resize();
+    }
+
+    togglePanel(button, panel) {
+        if (panel.hasClass('d-none')) {
+            this.showPanel(button, panel);
+        } else {
+            this.hidePanel(button, panel);
+        }
+        this.updateState();
+    }
+
+    override getCurrentState() {
+        const options = this.getEffectiveOptions();
+        const state: MonacoPaneState & ToolState = {
+            ...super.getCurrentState(),
+            wrap: options.wrap,
+            toolId: this.toolId,
+            args: this.getInputArgs(),
+            stdin: this.getInputStdin(),
+            stdinPanelShown: (this.monacoStdin && this.monacoEditorOpen) || !this.panelStdin.hasClass('d-none'),
+            monacoStdin: this.monacoStdin,
+            monacoEditorOpen: this.monacoEditorOpen,
+            monacoEditorHasBeenAutoOpened: this.monacoEditorHasBeenAutoOpened,
+            argsPanelShown: !this.panelArgs.hasClass('d-none'),
+        };
+        return state as MonacoPaneState;
+    }
+
+    setLanguage(languageId: false | string) {
+        if (languageId) {
+            this.options.enableToggle('wrap', false);
+            monaco.editor.setModelLanguage(unwrap(this.editor.getModel()), languageId);
+            this.editor.setValue('');
+            this.fontScale.setTarget(this.editor);
+            $(this.plainContentRoot).hide();
+            $(this.editorContentRoot).show();
+        } else {
+            this.options.enableToggle('wrap', true);
+            this.plainContentRoot.empty();
+            this.fontScale.setTarget('.content');
+            $(this.editorContentRoot).hide();
+            $(this.plainContentRoot).show();
+        }
+    }
+
+    clickableUrls(text) {
+        return text.replace(
+            // URL detection regex grabbed from https://stackoverflow.com/a/3809435
+            /(https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*))/,
+            '<a href="$1" target="_blank">$1</a>'
+        );
+    }
+
+    override onCompiler(compilerId: number, compiler: any, options: string, editorId: number, treeId: number) {
+        // TODO(jeremy-rifkin): This should probably be done in the base pane / standard across all panes
+        if (this.compilerInfo.compilerId !== compilerId) return;
+        this.compilerInfo.compilerName = compiler ? compiler.name : '';
+        this.compilerInfo.editorId = editorId;
+        this.compilerInfo.treeId = treeId;
+        this.updateTitle();
+    }
+
+    onCompileResult(id, compiler, result) {
+        try {
+            if (id !== this.compilerInfo.compilerId) return;
+            if (compiler) this.compilerInfo.compilerName = compiler.name;
+
+            const foundTool = _.find(compiler.tools, tool => tool.tool.id === this.toolId);
```

Comment:
Wasn't confident enough to try to remove the `_.find` uses here, wanted to minimize potential problems introduced by this pr